libc/stdio/stdio_indicator.c

```diff
@@ -0,0 +1,630 @@
+
+	{
+		long pos = 0;
+		errno = 0;
+		while (pos < 10 * testdata_hugeSize) {
+			TEST_ASSERT_EQUAL_INT(0, fseek(f, pos, SEEK_SET));
+			int errnoBefore = errno;
+			TEST_ASSERT_EQUAL_INT(0, feof(f));
+			TEST_ASSERT_EQUAL(errnoBefore, errno);
+			pos += 97;
+			errno++;
+		}
+	}
+
+	remove("errno_preserve");
+	fclose(f);
+}
+
+
+TEST_GROUP_RUNNER(stdio_feof)
+{
+	RUN_TEST_CASE(stdio_feof, not_empty_all_modes);
+	RUN_TEST_CASE(stdio_feof, empty_all_modes);
+	RUN_TEST_CASE(stdio_feof, preserve_errno_huge_size)
+}
+
+
+TEST_GROUP(stdio_ftell);
+
+TEST_SETUP(stdio_ftell)
+{
+}
+
+
+TEST_TEAR_DOWN(stdio_ftell)
+{
+}
+
+
+TEST(stdio_ftell, correct_position_not_empty)
+{
+	FILE *f;
+
+	for (int i = 0; i < sizeof(modes) / sizeof(char *); ++i) {
+
+		char filename[FNAME_LEN];
+		TEST_ASSERT_GREATER_OR_EQUAL(0, sprintf(filename, "file_FTELL_filled_%s", modes[i]));
+
+		f = fopen(filename, "w+");
+		char *data = testdata_createCharStr(255);
+		TEST_ASSERT_NOT_EQUAL_INT(EOF, fputs(data, f));
+		free(data);
+		TEST_ASSERT_EQUAL_INT(0, fclose(f));
+
+		errno = 0;
+		TEST_ASSERT_EQUAL(-1, ftell(f));
+		TEST_ASSERT_EQUAL(ftell(f), ftello(f));
+		TEST_ASSERT_EQUAL(EBADF, errno);
+
+		f = fopen(filename, modes[i]);
+
+		TEST_ASSERT_NOT_NULL(f);
+		TEST_ASSERT_EQUAL(ftell(f), ftello(f));
+
+		if (EQ("w", modes[i]) || EQ("wb", modes[i])) { /* can't read in "w" mode */
+			TEST_ASSERT_EQUAL_INT(0, fclose(f));
+			TEST_ASSERT_EQUAL(0, remove(filename));
+			continue;
+		}
+
+		while (fgetc(f) != EOF)
+			continue;
+
+		if (EQ("w+", modes[i]) || EQ("wb+", modes[i]))
+			TEST_ASSERT_EQUAL_INT64_MESSAGE(0, ftell(f), modes[i]);
+		else
+			TEST_ASSERT_EQUAL_INT64_MESSAGE(254, ftell(f), modes[i]);
+
+		TEST_ASSERT_EQUAL(ftell(f), ftello(f));
+
+
+		if (EQ("w+", modes[i]) || EQ("wb+", modes[i])) {
+			TEST_ASSERT_EQUAL_MESSAGE(0, ftell(f), modes[i]);
+		}
+		else {
+			TEST_ASSERT_EQUAL_INT64_MESSAGE(254, ftell(f), modes[i]);
+			TEST_ASSERT_EQUAL_INT(0, fseek(f, 2, SEEK_END));
+			TEST_ASSERT_EQUAL(256, ftell(f));
+		}
+		TEST_ASSERT_EQUAL_INT64(ftell(f), ftello(f));
+
+
+		TEST_ASSERT_EQUAL_INT(0, fclose(f));
+		TEST_ASSERT_EQUAL_INT(0, remove(filename));
+	}
+}
+
+
+TEST(stdio_ftell, correct_position_empty)
+{
+	FILE *f;
+
+	for (int i = 0; i < sizeof(modes) / sizeof(char *); ++i) {
+
+		char filename[FNAME_LEN];
+		TEST_ASSERT_GREATER_OR_EQUAL(0, sprintf(filename, "file_FTELL_empty_%s", modes[i]));
+
+		f = fopen(filename, "w+");
+		TEST_ASSERT_EQUAL_INT(0, fclose(f));
+
+		f = fopen(filename, modes[i]);
+
+		TEST_ASSERT_EQUAL_INT64(0, ftell(f));
+
+		TEST_ASSERT_EQUAL_INT(0, remove(filename));
+	}
+}
+
+
+TEST(stdio_ftell, bad_file_descriptor)
+{
+	errno = 0;
+	char filename[] = "test_bfd.txt";
+	FILE *f = fopen(filename, "w+");
+	TEST_ASSERT_EQUAL_INT(0, fclose(f));
+	TEST_ASSERT_EQUAL(-1, ftell(f));
+	TEST_ASSERT_EQUAL_INT(EBADF, errno);
+	TEST_ASSERT_EQUAL_INT(0, remove(filename));
+}
+
+
+TEST(stdio_ftell, wrong_stream_type)
+{
+	TEST_IGNORE_MESSAGE("Issue 923");
```

Comment:
Let's use the following format as in the rest of our tests.

```suggestion
	TEST_IGNORE_MESSAGE("#923 issue");
```

Reply:
Moreover the issue is only about sockets, so you can enable checks for pipe and fifo.